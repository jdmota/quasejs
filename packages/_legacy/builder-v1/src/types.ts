import { BuilderUtil, ModuleContext } from "./plugins/context";

export type Transforms = readonly (readonly string[])[];

export type Data = Buffer | string | Uint8Array;

export type DataType = "buffer" | "string" | "uint8array";

export type Loc = {
  line: number;
  column?: number;
};

export type AstInfo = {
  type: string;
  version: string;
  isDirty: boolean;
  program: any;
};

export type TransformableAsset = {
  type: string;
  data: Data;
  ast: AstInfo | null;
  map: any | null;
  depsInfo: DepsInfo | null;
  meta: { [key: string]: any } | null;
};

export type GenerateOutput = {
  data: Data;
  map: any;
};

export type ImportedName = {
  request: string;
  imported: string;
  name: string;
  loc?: Loc;
};

export type ExportedName = {
  request?: string;
  imported?: string;
  name: string;
  loc?: Loc;
};

export type Dep = {
  path?: string; // Possibly provide an already resolved dependency
  loc?: Loc;
  async?: boolean;
  transforms?: Transforms;
};

export type InnerDep = {
  type: string;
  data: Data;
  loc?: Loc;
  async?: boolean;
  transforms?: Transforms;
};

export type DepsInfo = {
  dependencies?: Map<string, Dep>;
  innerDependencies?: Map<string, InnerDep>;
  importedNames?: ImportedName[];
  exportedNames?: ExportedName[];
};

export type WatchedFileInfo = { time: number; onlyExistance?: boolean };
export type WatchedFiles = Map<string, WatchedFileInfo>;

export type Manifest = {
  files: string[];
  moduleToAssets: Map<string, string[]>;
};

export type RuntimeManifest = {
  f: string[];
  m: { [key: string]: number[] };
};

export type ResolvedDep = Readonly<{
  id: string;
  request: string;
  path: string;
  async: boolean;
  transforms: Transforms;
}>;

export type ResolvedInnerDep = Readonly<{
  id: string;
  innerId: string;
  type: string;
  async: boolean;
  transforms: Transforms;
}>;

export type FinalModule = Readonly<{
  id: string;
  path: string;
  relativePath: string;
  innerId: string | null;
  type: string;
  transformedId: number;
  asset: SharedArrayBuffer;
  resolvedId: number;
  dependencies: ReadonlyMap<string, ResolvedDep>;
  innerDependencies: ReadonlyMap<string, ResolvedInnerDep>;
  requires: readonly (ResolvedDep | ResolvedInnerDep)[];
}>;

export type FinalAsset = {
  relativeDest: string;
  hash: string | null;
  isEntry: boolean;
  runtime: {
    code: string | null;
    manifest: RuntimeManifest | null;
  };
  manifest: Manifest;
  module: FinalModule;
  srcs: Map<string, FinalModule>;
  inlineAssets: FinalAsset[];
};

export type ProcessedGraph = Readonly<{
  hashIds: ReadonlyMap<string, string>;
  moduleToFile: ReadonlyMap<FinalModule, FinalAsset>;
  files: readonly FinalAsset[];
  moduleToAssets: Readonly<{ [id: string]: string[] }>;
}>;

export type ToWrite = {
  data: Data;
  map?: any;
};

export type PerformanceOpts = {
  hints: boolean | "warning" | "error";
  maxEntrypointSize: number;
  maxAssetSize: number;
  assetFilter: (asset: string) => boolean;
};

export type Info = {
  moduleId: string;
  file: string;
  hash: string | null;
  size: number;
  isEntry: boolean;
};

export type HmrUpdate = Readonly<{
  updates: Readonly<{
    id: string;
    file: string | null;
    prevFile: string | null;
    reloadApp: boolean;
  }>[];
  moduleToAssets: Readonly<{ [id: string]: string[] }>;
}>;

export type HmrMessage =
  | {
      type: "update";
      update: HmrUpdate;
    }
  | {
      type: "error";
      error: string;
    };

export type Output = {
  filesInfo: Info[];
  removedCount: number;
  time: number;
  timeCheckpoints?: Map<string, number>;
  hmrUpdate: HmrUpdate;
};

type MaybeAsync<T> = T | Promise<T>;
type MaybeAsyncOptional<T> = MaybeAsync<T | null | undefined>;

export type ProvidedPlugin<T> = void | string | T | [string | T, any];

export type ProvidedPluginsArr<T> = readonly ProvidedPlugin<T>[];

export type WarnCb = (msg: string) => void;
export type ErrorCb = (
  id: string,
  msg: string,
  code: string | null,
  loc: Loc | null
) => void;

export type Resolver = {
  name?: string;
  options?(options: any): MaybeAsync<any>;
  resolve(
    options: any,
    imported: string,
    ctx: ModuleContext
  ): MaybeAsyncOptional<
    string | false | { path: string; transforms?: Transforms }
  >;
};

export type Transformer = {
  name?: string;
  options?(options: any): MaybeAsync<any>;
  canReuseAST?: (options: any, ast: AstInfo) => boolean;
  parse?: (
    options: any,
    asset: TransformableAsset,
    ctx: ModuleContext
  ) => MaybeAsync<AstInfo>;
  transform?: (
    options: any,
    asset: TransformableAsset,
    ctx: ModuleContext
  ) => MaybeAsync<TransformableAsset>;
  generate?: (
    options: any,
    asset: TransformableAsset,
    ctx: ModuleContext
  ) => MaybeAsync<GenerateOutput>;
};

export type ICheckerImpl = {
  newModule(module: FinalModule): void;
  deletedModule(id: string): void;
  check(): MaybeAsync<void>;
};

export type Checker = {
  name?: string;
  options?(options: any): MaybeAsync<any>;
  checker(options: any, _: { warn: WarnCb; error: ErrorCb }): ICheckerImpl;
};

export type Packager = {
  name?: string;
  options?(options: any): MaybeAsync<any>;
  pack(
    options: any,
    asset: FinalAsset,
    inlines: ReadonlyMap<FinalAsset, ToWrite>,
    hashIds: ReadonlyMap<string, string>,
    ctx: BuilderUtil
  ): MaybeAsyncOptional<ToWrite>;
};

export type Plugin<T> = {
  name: string;
  plugin: T;
  options: any;
};

export type OptimizationOptions = {
  hashId: boolean;
  hashing: boolean;
  sourceMaps: boolean | "inline";
  minify: boolean;
  /* concatenateModules: boolean;
  treeshake: boolean; */
};

export type UserConfigOpts = {
  cwd: string;
  optimization: OptimizationOptions;
  resolvers: ProvidedPluginsArr<string>;
  transformers: ProvidedPluginsArr<string>;
  checkers: ProvidedPluginsArr<string>;
  packagers: ProvidedPluginsArr<string>;
};

export type Options = {
  mode: "production" | "development";
  context: string;
  entries: string[];
  dest: string;
  cwd: string;
  publicPath: string;
  dotGraph: string | null;
  runtime: {
    browser: boolean;
    node: boolean;
    worker: boolean;
  };
  codeFrameOptions: any;
  watch: boolean;
  watchOptions: any;
  hmr: boolean;
  performance: PerformanceOpts;
  optimization: OptimizationOptions;
  serviceWorker: any;
  reporter: ProvidedPlugin<Function>;
  resolvers: ProvidedPluginsArr<string>;
  transformers: ProvidedPluginsArr<string>;
  checkers: ProvidedPluginsArr<string>;
  packagers: ProvidedPluginsArr<string>;
  _debug: boolean;
};
