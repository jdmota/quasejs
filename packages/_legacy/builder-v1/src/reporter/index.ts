import { relative } from "../utils/path";
import { Output } from "../types";
import { Builder } from "../builder/builder";
import Logger from "./logger";

const prettyBytes = require("pretty-bytes");

export default class Reporter extends Logger {
  constructor(options = {}, builder: Builder) {
    super(options);

    builder.on("status", message => {
      this.progress(message);
    });

    builder.on(
      "build-success",
      ({ filesInfo, removedCount, time, timeCheckpoints }: Output) => {
        const COLUMNS: { align: "left" | "right" }[] = [
          { align: "left" }, // isEntry
          { align: "left" }, // name
          { align: "right" }, // size
          { align: "left" }, // performance message
        ];

        const { performance, dest } = builder.options;
        const table: (string | number)[][] = [];

        for (const { file, size, isEntry } of filesInfo) {
          if (performance.assetFilter(file)) {
            let message = "";
            if (performance.hints) {
              if (isEntry && size > performance.maxEntrypointSize) {
                message = ` > ${prettyBytes(
                  performance.maxEntrypointSize
                )} [performance!]`;
              } else if (size > performance.maxAssetSize) {
                message = ` > ${prettyBytes(
                  performance.maxAssetSize
                )} [performance!]`;
              }
            }

            table.push([
              isEntry ? "[entry]" : "",
              relative(file, dest),
              prettyBytes(size),
              message,
            ]);
          }
        }

        this.log("");
        this.table(COLUMNS, table);
        this.log("");

        if (removedCount) {
          this.log(
            `Removed ${removedCount} old file${
              removedCount === 1 ? "" : "s"
            }.\n`
          );
        }

        if (this.isTest) {
          this.success("Built!");
        } else {
          const timeStr =
            time < 1000 ? `${time}ms` : `${(time / 1000).toFixed(2)}s`;
          if (timeCheckpoints) {
            this.table([], Array.from(timeCheckpoints));
            this.log("");
          }
          this.success(`Built in ${timeStr}!`);
        }
      }
    );

    builder.on("build-error", (err: string | Error) => this.error(err));

    builder.on("build-cancelled", () => {
      this.progress("Previous build cancelled...");
    });

    builder.on("watching", (files: string[]) => {
      this.progress(`Watching ${files.length} files...`);
    });

    builder.on("updates", (updates: { path: string; type: string }[]) => {
      this.clear();
      for (const { path, type } of updates) {
        this.info(`File ${relative(path, builder.options.cwd)} was ${type}.`);
      }
    });

    builder.on("warning", (w: string | Error) => this.warn(w));

    builder.on("hmr-starting", () => {
      this.progress("HMR server starting...");
    });

    builder.on(
      "hmr-started",
      ({ hostname, port }: { hostname: string; port: number }) => {
        this.persistent(`HMR server listening at ${hostname}:${port}...`);
      }
    );

    builder.on("hmr-error", (w: string | Error) => this.warn(w));

    builder.on("sigint", () => {
      this.stopSpinner();
      this.info("Closing...");
    });
  }
}
