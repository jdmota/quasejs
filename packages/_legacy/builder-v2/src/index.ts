import { Options } from "./types";
import { Builder } from "./builder/builder";
import Reporter from "./reporter";

export async function cli(options: Options) {
  const builder = new Builder(options);
  const reporter = new Reporter({}, builder); // TODO custom reporter and options

  process.once("SIGINT", () => {
    builder.emit("sigint");
    builder.stop();
  });

  const { result, firstJob } = await builder.start();

  return {
    reporter,
    builder,
    setupResult: result,
    firstJob,
  };
}

export { Builder };

export { schema, normalizeOptions } from "./options";
