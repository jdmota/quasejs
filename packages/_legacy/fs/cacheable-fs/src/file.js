// @flow
const fs = require( "fs-extra" );

export default class File {

  +location: string;
  _stat: any;
  _read: any;
  _readdir: any;

  constructor( location: string ) {
    this.location = location;
    this._stat = null;
    this._read = null;
    this._readdir = null;
  }

  async stat() {
    return this._stat || ( this._stat = fs.stat( this.location ) );
  }

  async readFile( encoding: ?string ) {
    const read = this._read || ( this._read = fs.readFile( this.location, encoding ) );
    const buffer = await read;
    return encoding ? buffer.toString( encoding ) : buffer;
  }

  async readdir() {
    return this._readdir || ( this._readdir = fs.readdir( this.location ) );
  }

  purgeContent() {
    this._read = null;
    this._readdir = null;
  }

}
