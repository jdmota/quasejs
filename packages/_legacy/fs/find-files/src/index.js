const fs = require( "fs-extra" );
const path = require( "path" );
const ignoredDirectories = require( "ignore-by-default" ).directories().map( dir => `!**/${dir}/**` );
const mm = require( "micromatch" );
const globParent = require( "glob-parent" );
const toAbsoluteGlob = require( "to-absolute-glob" );
const isNegatedGlob = require( "is-negated-glob" );
const slash = require( "slash" );
const Observable = require( "zen-observable" );

class Pattern {

  constructor( _pattern, { cwd, micromatch } ) {
    const micromatchOpts = this.negated ? Object.assign( {}, micromatch, { dot: true } ) : micromatch;
    const { negated, pattern } = isNegatedGlob( _pattern );
    this.negated = negated;
    this.regexp = mm.makeRe( pattern, micromatchOpts );
    this.endsWithGlobstar = pattern.slice( -3 ) === "/**";
    this.parent = globParent( pattern ).replace( /^\.$/g, "" );
    this.fullParent = toAbsoluteGlob( this.parent, { cwd } );
  }

}

class Reader {

  constructor( patterns, observer, options ) {
    this.patterns = patterns;
    this.cwd = options.cwd;
    this.observer = observer;
    this.cancelled = false;
    this.pending = 0;
    this.includeDirs = options.includeDirs;
    this.visitedDirs = options.visitedDirs;
    this.relative = options.relative;
    this.fsStat = options.fs.stat;
    this.fsReadDir = options.fs.readdir;
    this.error = this.error.bind( this );
    this.cancel = this.cancel.bind( this );
  }

  start() {
    for ( const { negated, fullParent } of this.patterns ) {
      if ( !negated ) {
        this.check( fullParent );
      }
    }
    this.checkPending();
    return this.cancel;
  }

  checkPending() {
    if ( this.pending === 0 ) {
      this.complete();
    }
  }

  next( file ) {
    this.observer.next( file );
  }

  complete() {
    this.observer.complete();
  }

  error( error ) {
    this.observer.error( error );
  }

  cancel() {
    this.cancelled = true;
  }

  // We start from the end
  // negated matches
  //    0      0     keep going
  //    0      1     return true
  //    1      0     keep going
  //    1      1     return false
  matches( string ) {
    const patterns = this.patterns;
    let i = patterns.length;
    while ( i-- ) {
      const negated = patterns[ i ].negated;
      const matches = patterns[ i ].regexp.test( string );
      if ( matches ) {
        return !negated;
      }
    }
    return false;
  }

  matchesDir( dir ) {
    const patterns = this.patterns;
    let i = patterns.length;
    while ( i-- ) {
      const p = patterns[ i ];
      if ( p.negated ) {
        if ( p.endsWithGlobstar ) {
          const matches = p.regexp.test( dir + "/" );
          if ( matches ) {
            return false;
          }
        }
      } else {
        const parent = p.parent;
        let matches = true;
        const len = Math.min( parent.length, dir.length );
        for ( let i = 0; i < len; i++ ) {
          if ( parent.charCodeAt( i ) !== dir.charCodeAt( i ) ) {
            matches = false;
            break;
          }
        }
        if ( matches ) {
          return true;
        }
      }
    }
    return false;
  }

  check( file ) {
    this.pending++;

    this.fsStat( file ).then( stats => {

      if ( this.cancelled ) {
        return;
      }

      this.pending--;

      const relativePath = slash( path.relative( this.cwd, file ) );

      if ( stats.isDirectory() ) {
        if ( this.includeDirs && this.matches( relativePath ) ) {
          this.next( this.relative ? relativePath : file );
          this.readDir( file );
        } else if ( this.matchesDir( relativePath ) ) {
          this.readDir( file );
        }
      } else if ( stats.isFile() ) {
        if ( this.matches( relativePath ) ) {
          this.next( this.relative ? relativePath : file );
        }
      }

      this.checkPending();

    }, this.error );
  }

  readDir( folder ) {
    if ( this.visitedDirs ) {
      this.visitedDirs.push( slash( path.relative( this.cwd, folder ) ) );
    }

    this.pending++;

    this.fsReadDir( folder ).then( files => {

      if ( this.cancelled ) {
        return;
      }

      this.pending--;

      files.forEach( f => this.check( path.join( folder, f ) ) );

      this.checkPending();

    }, this.error );
  }

}

function makeSureArray( val ) {
  if ( val == null ) {
    return [];
  }
  return Array.isArray( val ) ? val : [ val ];
}

export function findFilesObservable( src, _opts ) {

  const options = Object.assign( {}, _opts );

  options.cwd = options.cwd ? path.resolve( options.cwd ) : process.cwd();
  options.fs = options.fs || fs;
  options.fs.stat = options.fs.stat || fs.stat;
  options.fs.readdir = options.fs.readdir || fs.readdir;
  options.append = makeSureArray( options.append || ignoredDirectories );

  const patterns = makeSureArray( src ).concat( options.append ).map(
    pattern => new Pattern( pattern, options )
  );

  return new Observable( observer => new Reader( patterns, observer, options ).start() );
}

export default function( patterns, options ) {
  return new Promise( ( resolve, reject ) => {
    const files = [];

    findFilesObservable( patterns, options ).subscribe( {
      error: reject,
      next: x => files.push( x ),
      complete: () => resolve( files )
    } );
  } );
}
