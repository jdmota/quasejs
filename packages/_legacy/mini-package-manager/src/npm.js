// @flow

// Keep this updated with https://github.com/npm/npm/blob/latest/lib/config/cmd-list.js
// Last update: https://github.com/npm/npm/blob/8c77dde74a9d8f9007667cd1/lib/config/cmd-list.js

const commands = {
  audit: true,
  dedupe: true,
  ci: true,
  install: true,
  "install-ci-test": [ "install", "ci" ],
  "install-test": [ "install", "test" ],
  link: true,
  ls: true,
  outdated: true,
  prune: true,
  uninstall: true,
  update: true,
};

const shorthands = {
  un: "uninstall",
  list: "ls",
  ln: "link",
  i: "install",
  it: "install-test",
  cit: "install-ci-test",
  up: "update",
  ddp: "dedupe"
};

const affordances = {
  la: "ls",
  ll: "ls",
  ic: "ci",
  isntall: "install",
  "find-dupes": "dedupe",
  upgrade: "update",
  udpate: "update",
  add: "install",
  unlink: "uninstall",
  remove: "uninstall",
  rm: "uninstall",
  r: "uninstall",
  sit: "cit"
};

export function normalize( command: string ): boolean | string[] {
  command = affordances[ command ] || command;
  command = shorthands[ command ] || command;
  return commands[ command ] || false;
}

export const before = [
  "package.json",
  ".npmrc",
  ".yarnrc"
];

export const after = [
  "package-lock.json",
  "npm-shrinkwrap.json",
  "yarn.lock",
  "shrinkwrap.yaml",
  "package.json",
  ".npmrc",
  ".yarnrc"
];
