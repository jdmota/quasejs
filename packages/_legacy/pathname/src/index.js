import basename from "./basename";
import changeBase from "./change-base";
import dirname from "./dirname";
import extname from "./extname";
import isAbsolute from "./is-absolute";
import join from "./join";
import match from "./match";
import normalize from "./normalize";
import pathToRegExp from "./path-to-regexp";
import relative from "./relative";
import resolve from "./resolve";

export default {
  basename,
  changeBase,
  dirname,
  extname,
  isAbsolute,
  join,
  match,
  normalize,
  pathToRegExp,
  relative,
  resolve
};
