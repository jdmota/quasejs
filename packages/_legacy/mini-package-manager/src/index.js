// @flow
import { execOnInternal, execOnFolder, execOnBoth, copyMaybe, crawl } from "./utils";
import { before as npmBefore, after as npmAfter, normalize as npmNormalize } from "./npm";

const path = require( "path" );
const fs = require( "fs-extra" );
const crypto = require( "crypto" );
const homedir = require( "os" ).homedir();
const readPkgUp = require( "read-pkg-up" );

function connectNodeModules( deps: Set<string>, fromBase: string, toBase: string ) {
  const promises = [];

  for ( const dep of deps ) {

    const _from = path.join( fromBase, "node_modules", dep );
    const to = path.join( toBase, "node_modules", dep );

    promises.push(
      fs.pathExists( _from ).then( exists => {
        if ( exists ) {
          return crawl( _from, item => {
            if ( item.stats.isFile() ) {
              const relative = path.relative( _from, item.path );
              const dest = path.resolve( to, relative );
              return (
                /\.js$/.test( dest ) || !/\..+$/.test( dest ) ?
                  fs.outputFile( dest, `module.exports=require(${JSON.stringify( item.path )});` ) :
                  fs.copy( item.path, dest )
              );
            }
          } );
        }
      } )
    );
  }

  return Promise.all( promises );
}

function exec( command: string, args: string[], folder: string, internal: string ) {

  const thing = npmNormalize( command );

  if ( Array.isArray( thing ) ) {
    return execOnBoth( thing, args, folder, internal );
  }

  if ( thing ) {
    return execOnInternal( command, args, folder, internal );
  }

  return execOnFolder( command, args, folder );
}

// TODO https://stackoverflow.com/questions/14742553/npm-local-install-package-to-custom-location
// TODO https://github.com/yarnpkg/yarn/issues/1684

// modules in ./npm_modules/node_modules
// npm install slash --prefix ./npm_modules

// modules in ./yarn_modules
// yarn add slash --modules-folder ./yarn_modules

export default async function( command: string, args: string[] ) {

  const { pkg, path: pkgPath } = readPkgUp();

  const folder = path.join( pkgPath, ".." );
  const folderName = path.basename( folder );

  const refPath = path.join( folder, ".quase-mini-package-manager" );

  let refExisted = false;
  let ref;

  try {
    ref = await fs.readFile( refPath, "utf8" );
    refExisted = true;
  } catch ( err ) {
    if ( err.code === "ENOENT" ) {
      ref = `${folderName}-${crypto.createHash( "md5" ).update( Date.now() + "", "utf8" ).digest( "hex" )}`;
    } else {
      throw err;
    }
  }

  const internal = path.join( homedir, "quase-mini-package-manager", ref );

  await fs.ensureDir( internal );

  await copyMaybe( folder, internal, npmBefore );

  if ( !refExisted ) {
    await fs.writeFile( refPath, ref );
  }

  await exec( command, args, folder, internal );

  await copyMaybe( internal, folder, npmAfter );

  const deps = new Set();

  for ( const kind of [ "dependencies", "devDependencies", "peerDependencies", "optionalDependencies" ] ) {
    for ( const name in Object( pkg[ kind ] ) ) {
      deps.add( name );
    }
  }

  await connectNodeModules( deps, internal, folder );

}
