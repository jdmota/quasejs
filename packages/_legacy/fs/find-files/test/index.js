import { findFilesObservable } from "../src";

const path = require( "path" );

function run( patterns ) {
  return new Promise( ( resolve, reject ) => {
    const files = [];
    const visitedDirs = [];

    findFilesObservable( patterns, {
      cwd: path.join( __dirname, "fixtures" ),
      relative: true,
      visitedDirs
    } ).subscribe( {
      error: reject,
      next: x => files.push( x ),
      complete: () => resolve( {
        files: files.sort(),
        visitedDirs: visitedDirs.sort()
      } )
    } );
  } );
}

it( "basic", async() => {

  expect(
    await run( [ "a/**", "!b/**", "c/**" ] )
  ).toMatchSnapshot();

} );
