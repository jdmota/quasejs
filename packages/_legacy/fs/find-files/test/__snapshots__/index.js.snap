// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
Object {
  "files": Array [
    "a/1.js",
    "a/2.js",
    "c/1.js",
    "c/2.js",
  ],
  "visitedDirs": Array [
    "a",
    "c",
  ],
}
`;
